BookingsApi/BookingsApi.DAL/Queries/GetJudiciaryPersonByExternalRefIdQuery.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Threading.Tasks;
+using BookingsApi.Domain;
+using BookingsApi.DAL.Queries.Core;
+using Microsoft.EntityFrameworkCore;
+
+namespace BookingsApi.DAL.Queries
+{
+    public class GetJudiciaryPersonByExternalRefIdQuery : IQuery
+    {
+        public Guid ExternalRefId { get; set; }
+
+        public GetJudiciaryPersonByExternalRefIdQuery(Guid externalRefId)
+        {
+            ExternalRefId = externalRefId;
+        }
+    }
+    
+    public class GetJudiciaryPersonByExternalRefIdQueryHandler : IQueryHandler<GetJudiciaryPersonByExternalRefIdQuery, JudiciaryPerson>
+    {
+        private readonly BookingsDbContext _context;
+
+        public GetJudiciaryPersonByExternalRefIdQueryHandler(BookingsDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<JudiciaryPerson> Handle(GetJudiciaryPersonByExternalRefIdQuery query)
+        {
+            return await _context.JudiciaryPersons.SingleOrDefaultAsync(x => x.ExternalRefId == query.ExternalRefId);   
```

Comment:
could return the task (remove `async` and `await`)

Reply:
Not sure why we want to do that?

Reply:
I can think of a couple of top of my head why having less unnecessary awaits is better..
by default configureawait is true so it will wait for the calling thread to be available before resuming. If there is another await after this one the thread may be nabbed and would take slightly longer to resume execution?
also If moving to configureawait its one less place to think about
Its also bringing no benefit awaiting there
my personal preference is also not to have a return on the same line as await but i know thats controversial but still, 2 birds 1 stone :P
i cant think of any reasons why keeping would be better?

Reply:
there is no thread switching on a Task.IsCompletedSuccessfully, it will process straight back to the calling code

Reply:
is there not, do you have a link handy so i can read about it further please :)

I found some guidelines on the matter, this guy suggests keeping the async await but has noted that it has performance implications. Im not sure many of his points apply to this scenario

https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task

Happy to mark as resolved